### PR TITLE
Add missing instantiation for PETSc with complex values

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes https://cdash.43-1.org/viewBuildError.php?buildid=7282.

### CODE REVIEW

File: source/lac/affine_constraints.cc [hunk 1]
```diff
@@ -109,6 +109,14 @@ INSTANTIATE_DLTG_BLOCK_VECTORMATRIX(PETScWrappers::MPI::BlockSparseMatrix,
 INSTANTIATE_DLTG_MATRIX(PETScWrappers::SparseMatrix);
 INSTANTIATE_DLTG_MATRIX(PETScWrappers::MPI::SparseMatrix);
 INSTANTIATE_DLTG_MATRIX(PETScWrappers::MPI::BlockSparseMatrix);
+#  ifndef DOXYGEN
+#    ifdef DEAL_II_PETSC_WITH_COMPLEX
+template void
+dealii::AffineConstraints<double>::distribute<
```
Comment (bangerth) [inline source/lac/affine_constraints.cc:115]: Don't you want this to be `PetscScalar`? Or do I misunderstand why you're guarding this with `DEAL_II_PETSC_WITH_COMPLEX`? (Maybe a comment would help :-) )

Comment (masterleinad) [reply source/lac/affine_constraints.cc:115]: https://cdash.43-1.org/viewBuildError.php?buildid=7282 says that we need this combination. I didn't investigate more where that comes from.

Comment (bangerth) [reply source/lac/affine_constraints.cc:115]: I see. Never mind then!

**Pull request merged**